esbuild.config.mjs

```diff
@@ -0,0 +1,75 @@
+import * as esbuild from 'esbuild';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+import { fileURLToPath } from 'url';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const srcDir = path.join(__dirname, 'src');
+const actionsDir = path.join(__dirname, 'actions');
+const buildDir = path.join(__dirname, 'build'); // Build output directory
+
+// Shared esbuild options
+const sharedEsbuildOptions = {
+  bundle: true,       // Bundle dependencies
+  platform: 'node',   // Target Node.js environment
+  target: 'node20',   // Target Node.js version (must match action.yml)
```

Comment:
this can be lower to the target version. so instead of "must match action.yml", it's "must not be greater than action.yml"

keep it to node20 is very likely to make it always working through the whole life cycle of this script.